core/src/test/java/org/jdbi/v3/core/mapper/ImmutablesTest.java

```diff
@@ -0,0 +1,131 @@
+        h.registerRowMapper(ImmutablesMapperFactory.mapImmutable(Train.class, ImmutableTrain.class, ImmutableTrain::builder));
+
+        assertThat(
+            h.createUpdate("insert into train(name, carriages, observation_car) values (:name, :carriages, :observationCar)")
+                .bindProperties(ImmutableTrain.builder().name("Zephyr").carriages(8).observationCar(true).build())
+                .execute())
+            .isEqualTo(1);
+
+        assertThat(
+            h.createQuery("select * from train")
+                .mapTo(Train.class)
+                .findOnly())
+            .extracting("name", "carriages", "observationCar")
+            .containsExactly("Zephyr", 8, true);
+    }
+    // end::example[]
+
+    @Test
+    public void parameterizedTest() {
+        assertThat(
+            h.createUpdate("insert into immutables(t, x) values (:t, :x)")
+                .bindProperties(ImmutableSubValue.<String, Integer>builder().t(42).x("foo").build())
+                .execute())
+            .isEqualTo(1);
+
+        assertThat(
+            h.createQuery("select * from immutables")
+                .mapTo(new GenericType<SubValue<String, Integer>>() {})
+                .findOnly())
+            .extracting("t", "x")
+            .containsExactly(42, "foo");
+    }
+
+    public interface BaseValue<T> {
+        T t();
+    }
+
+    @Value.Immutable
+    public interface SubValue<X, T> extends BaseValue<T> {
+        X x();
+    }
+
+    @Value.Immutable
+    @Value.Modifiable
+    public interface FooBarBaz {
+        int id();
+        Optional<String> foo();
+        OptionalInt bar();
+        OptionalDouble baz();
+    }
+
+    @Test
+    public void testModifiable() {
+        h.registerRowMapper(ImmutablesMapperFactory.mapImmutable(FooBarBaz.class, ImmutableFooBarBaz.class, ImmutableFooBarBaz::builder));
+        h.registerRowMapper(ImmutablesMapperFactory.mapModifiable(FooBarBaz.class, ModifiableFooBarBaz.class, ModifiableFooBarBaz::create));
+        h.execute("create table fbb (id serial, foo varchar, bar int, baz real)");
+
+        assertThat(h.createUpdate("insert into fbb (id, foo, bar, baz) values (:id, :foo, :bar, :baz)")
+                .bindProperties(ModifiableFooBarBaz.create().setFoo("foo").setBar(42).setBaz(1.0))
+                .execute())
+            .isEqualTo(1);
+
+        assertThat(h.createQuery("select * from fbb")
+                .mapTo(ModifiableFooBarBaz.class)
+                .findOnly())
+            .extracting("id", "foo", "bar", "baz")
+            .containsExactly(1, Optional.of("foo"), OptionalInt.of(42), OptionalDouble.of(1.0));
+
+        assertThat(h.createQuery("select * from fbb")
+                .mapTo(ImmutableFooBarBaz.class)
+                .findOnly())
+            .extracting("id", "foo", "bar", "baz")
+        // FIXME: https://github.com/joel-costigliola/assertj-core/pull/1360
```

Comment:
This still an issue?

Reply:
It's merged but not released yet :/

Reply:
Pinged the maintainer asking for a release.

Reply:
Sounds like it's not going to happen for a bit, so we shouldn't block on it, unfortunately. https://github.com/joel-costigliola/assertj-core/pull/1360#issuecomment-451627976